spec/features/auth_spec.rb

```diff
@@ -16,4 +16,31 @@
     expect(page).to have_content("user@example.com")
   end
 
+  scenario 'Users can register and automatically login after registration' do
+    visit root_path
+
+    click_link "Register"
+
+    within(".registration-form") {expect(page).to have_content("Register")}
+
+    fill_in "Name", with: "Mahamdi Gandhi"
+    fill_in "Email", with: "peace@hotmail.com"
+    fill_in "Password", with: "lovenotwar"
+    fill_in "Confirm", with: "lovenotwar"
+
+    within(".registration-form") {click_on "Register"}
+    expect(page).to have_content("Welcome peace@hotmail.com")
+
+    # expect(page).to have_xpath?('/registrations/new')
+    #
+    # fill_in "Name", with: "John Smith"
+    # fill_in "Email", with: "johnsmith@yahoo.com"
+    # fill_in "Password", with: "johnsmith"
+    # fill_in "Confirm", with: "johnsmith"
+    # click_button('Register')
+    # expect(page).to have_content("Welcome John Smith")
+    # expect(page).to have_contect("You are logged in successfully.")
```

Comment:
Solid and correct solution, but don't check in commented out tests. When you're on a team the next dev will come along and say "why is this here? should it be uncommented?" etc